server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -63,26 +66,66 @@ pub fn insert_program_request_requisition(
 }
 
 fn validate(
-    connection: &StorageConnection,
-    store_id: &str,
+    ctx: &ServiceContext,
     input: &InsertProgramRequestRequisition,
-) -> Result<(), OutError> {
+) -> Result<(ProgramRow, ProgramRequisitionOrderTypeRow), OutError> {
+    let connection = &ctx.connection;
+
     if let Some(_) = check_requisition_exists(connection, &input.id)? {
         return Err(OutError::RequisitionAlreadyExists);
     }
 
-    // TODO: Check if we already have reached out max requisitions for this period
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let requisitions = RequisitionRepository::new(&connection).query_by_filter(
+        RequisitionFilter::new().order_type(EqualFilter::equal_to(&input.program_order_type_id)),
```

Comment:
I'm not getting any requisitions returned by this filter. Which results in duplicates for the period+program combination:

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/9192912/235811013-1cd34ba8-8716-4931-8d1e-514f3a9da948.png">

🤔

Reply:
Will investigate 🤔  I only tested with tests...